automated/linux/block-device/block-device.sh

```diff
@@ -0,0 +1,229 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0-only
+# Copyright (C) 2019 Linaro Ltd.
+
+# shellcheck disable=SC1091
+. ../../lib/sh-test-lib
+OUTPUT="$(pwd)/output"
+RESULT_FILE="${OUTPUT}/result.txt"
+
+PARTDEV=""
+PARTNAME=""
+DEVICE=""
+DEVNAME=""
+
+SKIP_INSTALL="false"
+FORMAT_DEVICE="false"
+
+MOUNT_DIR=$(mktemp -d "/tmp/block-device.XXXXX")
+TMPFILE=$(mktemp "/tmp/10M.XXXXX")
+RESULT=pass
+
+usage() {
+	echo "\
+	Usage: $0
+		     [-d </dev/sdXN>]
+		     [-s  <true|false>]
+		     [-f ]
+
+	-d </dev/sdXN>
+		This is the block device to be tested, eg. /dev/sda1.
+		It should be a partition, not a raw block device
+	-f
+		This will erase the device and create a new partition table
+		with a single ext4 partition on it.
+	-s <true|false>
+		Tell the test to skip installation of dependencies, or not
+
+	This test will perform tests on a block device.
+	"
+}
+
+while getopts "d:fs:h" opts; do
+	case "$opts" in
+		d) PARTDEV="${OPTARG}";
+		   PARTNAME=$(basename -- "${PARTDEV}");
+		   DEVICE=${PARTDEV/%[[:digit:]]/}
+		   DEVNAME=$(basename -- "${DEVICE}")
+		   ;;
+		f) FORMAT_DEVICE="true" ;;
```

Comment:
I think this is wrong. At least this isn't in line with the YAML file, where it's always present.

Reply:
Ah yes, I've started to convert it from "-f value" to just "-f" and not completed the change. But it needs to have the value for the YAML to make sense. I'll fix...